pipeline/workflow-builder.Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM quay.io/kiegroup/kogito-swf-builder-nightly:main-2024-01-24 AS builder
+FROM quay.io/kiegroup/kogito-swf-builder-nightly:latest AS builder
```

Comment:
@rgolangh @masayag what do you think? Should we use version or `latest`?
We can always find the version based on the data the image was built

Reply:
I'd pin it to a specific version and not to the latest. 
It will allow us to stabilize current versions and periodically move on until there will be a release of it.